pallets/gear/src/internal.rs

```diff
@@ -310,12 +310,22 @@ where
             unreachable!("Failed to figure out correct wait hold bound");
         }
 
+        // TODO: remove, once duration-control added inside programs (#1173).
+        //
+        // This need in async scenarios, while we send gasless message and it
+        // goes into waitlist: then all funds become locked for storing in
+        // waitlist, instead of distribute for execution and storing.
+        let hold = HoldBound::<T>::by(CostsPerBlockOf::<T>::waitlist())
+            .duration(hold.expected_duration().min(500u32.unique_saturated_into()));
```

Comment:
As mentioned in comment above:
> // TODO: remove, once duration-control added inside programs (1173).
         //
         // This need in async scenarios, while we send gasless message and it
         // goes into waitlist: then all funds become locked for storing in
         // waitlist, instead of distribute for execution and storing.
         
It will be removed asap. It's just temporary workaround due to absence of other methods to solve the problem of locking all gas for gasless sending.